src/main/java/edu/kit/ipd/crowdcontrol/objectservice/notification/SQLEmailNotificationPolicy.java

```diff
@@ -0,0 +1,116 @@
+     */
+    public SQLEmailNotificationPolicy(MailSender mailSender, NotificationOperations operations) {
+        super(operations);
+        this.mailSender = mailSender;
+        this.operations = operations;
+    }
+
+    private boolean resultHasIdAndTokenField(Result<Record> result) {
+        Field<Integer> idField = DSL.field("id", Integer.class);
+        Field<String> tokenField = DSL.field("token", String.class);
+
+        return (result.get(0).field(idField) != null) && (result.get(0).field(tokenField) != null);
+    }
+
+    @Override
+    protected List<String> check(Notification notification) {
+        Result<Record> result = operations.runReadOnlySQL(notification.getQuery());
+        if (result.isNotEmpty()) {
+            if (resultHasIdAndTokenField(result)) {
+                // the result has the required columns, so we can bring it in a handier format
+                Map<Integer, NotificationTokenRecord> tokenRecords = result.stream()
+                        .map(record -> new NotificationTokenRecord(null, record.getValue("id", Integer.class),
+                                record.getValue("token", String.class), notification.getId())
+                        )
+                        .collect(Collectors.toMap(NotificationTokenRecord::getResultId, Function.identity()));
+
+                List<String> newTokenList = operations.diffTokenRecords(tokenRecords, notification.getId()).stream()
+                        .map(NotificationTokenRecord::getResultToken)
+                        .collect(Collectors.toList());
+                if (!newTokenList.isEmpty()) {
+                    return newTokenList;
+                }
+            } else {
+                // the result has any other form and we cannot provide any tokens
+                return Collections.emptyList();
+            }
+        }
+        return null;
+    }
+
+    @Override
+    protected void send(Notification notification, List<String> tokens) {
+        StringBuilder message = new StringBuilder();
+
+        // TODO fancy pancy html + css styled body with beautiful design and chocolate flavor ...if there is time
+        if (tokens.isEmpty()) {
+            message.append(notification.getDescription());
+        } else {
+            // search for placeholders in description
+            Map<String, String> placeholderMap = new HashMap<>();
+            placeholderMap.put("tokens", concatTokens(tokens));
+            try {
+                message.append(Template.apply(notification.getDescription(), placeholderMap));
+            } catch (IllegalArgumentException iae) {
+                message.append(notification.getDescription());
+            }
+        }
+
+        String subject = "[CrowdControl] " + notification.getName();
+        try {
+            for (String receiver : notification.getReceiverEmails())
+            mailSender.sendMail(receiver, subject, message.toString());
```

Comment:
we are in a curly braces land..., i would also catch the exception for each sendMail operation.